assignment1/string_manipulation.rb

```diff
@@ -0,0 +1,58 @@
+str1 = "RUBY parses a file by looking for <br/> one of the special tags that tells it to start interpreting the text as RUBY code. The parser then executes all of the code it finds until it runs into a RUBY closing <br/> tag.";
+str2 = "RUBY does not require (or support) explicit type definition in variable declaration; a variable's type is determined by the context in which the variable is used.";
+
+#a
+puts "a. Find occurrence of RUBY from string 1?"
+
+puts str1.scan("RUBY").count
+
+puts '
+b. Find the position where RUBY occurs in the string 1.'
+
+puts (0..str1.length).find_all { |i| str1[i,4] == 'RUBY'}
+
+puts '
+c. Create array of words in string 1 & print them using a recursive function?'
+split_str1 = str1.split(' ')
+split_str1.each { |i| puts split_str1[i] }
+
+puts '
+d. Capitalise string 1'
+
+puts split_str1.map { |word| word.capitalize }.join(' ')
+
+puts '
+e. Combine string 1 & 2?'
+
+puts str1 + str2
+
+puts '
+f. Cut the string 1 into 4 parts & print it.'
+
+puts str1.chars.each_slice(str1.length/3).map(&:join)
```

Comment:
this solution is not returning correct output
[222] pry(main)> "ABC".chars.each_slice(str1.length/3).map(&:join)
=> ["ABC"]
[223] pry(main)> "ABCABCABC".chars.each_slice(str1.length/3).map(&:join)
=> ["ABCABCABC"]
[224] pry(main)> "ABCABCABCD".chars.each_slice(str1.length/3).map(&:join)
=> ["ABCABCABCD"]

It should return four strings in array, Please fix

Reply:
resolved